tools/smithy-rs-tool-common/src/git.rs

```diff
@@ -16,26 +17,15 @@ pub use get_current_tag::GetCurrentTag;
 mod get_last_commit;
 pub use get_last_commit::GetLastCommit;
 
+mod get_repo_root;
+pub use get_repo_root::GetRepoRoot;
+
 /// Attempts to find git repository root from the given location.
 pub fn find_git_repository_root(repo_name: &str, location: impl AsRef<Path>) -> Result<PathBuf> {
```

Comment:
Is `repo_name` useful? Could we remove it? I see it's only called with either `aws-sdk-rust` or `smithy-rs`, and it is checked that the name of the directory coincides with what `git rev-parse --show-top-level` returns. I don't see the check of the `repo_name` coinciding with the directory name adding value.

I ask because I sometimes have `smithy-rs` checked out with in a dir not named `smithy-rs`.

Reply:
It's a fair point. The reason it's in here is we often checkout smithy-rs into aws-sdk-rust in CI so that we can use parts of both, and this is a sanity check that our CI script is set up correctly. It's also helpful in code to communicate which repository we're expecting to be working with. None of this is essential though.

Reply:
This is the patch I apply to my tree whenever I want to build the SDK https://github.com/awslabs/smithy-rs/pull/1158/commits/f76ad73a46ae1bf5b68c77ac4652838ae979785f It'd be nice if the script worked with any directory name.

Reply:
Made it log a warning instead of failing.